### DISCUSSION TITLE
OSM file as data

### DISCUSSION THREAD

**Discussion by SaeedDev94** [komoot/photon#713] [Q&A] [answered]
Hi, Thanks for this amazing project
Can i use (xml) OSM data file for photon data without using database (like postgres) at all ?
If the answer is yes how can i do that ?
The documents is not clear enough for me :(

**Comment by lonvia** [comment, marked as answer]
No, you can't. You either need to do an import with Nominatim into a postgres database first and create a Photon database from that or use our prepared databases.

**Comment by SaeedDev94** [comment]
@lonvia
I found `osm2pgsql/jammy 1.6.0+ds-1 amd64` in ubuntu repo via apt and installed it
I imported osm file via that but it seems the tables doesn't compatible with photon!
When i start photon for data import it fails because it needs `placex` table which i haven't it at all
Ubuntu `osm2pgsql` is out of date or I'm using it in wrong way?
`osm2pgsql -H localhost -U postgres -d nominatim -W map.osm`
(For now I'm using your database but it's pretty huge and I just trying to create a database limited to my country)

**Reply by lonvia** [reply]
Please reread my answer. I said you need to create a postgres database with **Nominatim**.

For country extracts read https://nominatim.org/2020/10/21/photon-country-extracts.html

**Reply by SaeedDev94** [reply]
@lonvia 
I was looking for a solution which didn't require compiling `Nominatim` C source code
I saw `osm2pgsql` in `Nominatim` before, So i thought i can import osm to PostgreSQL without using `nominate import ...` for `photon`
I wish `Nominatim` provide a pre-build (ready to use) as well as `photon` does ...
BTW i had to compile `Nominatim` and fortunately all went well
Thanks

**Reply by liangxu51** [reply]
@lonvia
If I download a country extract (prepared database) and setup photon with non-embedded elasticsearch server, how can I import the data into ES server? 
"java -jar photon" seems to only load the es-index from folder "photon_data" into the embedded ES server.
